routes/users.js

```diff
@@ -509,14 +509,39 @@ router.post('/checkout_cart', function(req, res, next) {
             if (err) return next(err);
           });
 
+          await User.findOne({ "_id": req.session.userId }, function(err, user) {
+            if (err) return next(err);
+
+            let report = user.report[0];
+            if (report === null | report === undefined) {
+              report = [];
+              report.push({});
+              report = report[0];
+            }
+
+            production_report = user.production_report;
+            if (production_report === null | production_report === undefined) {
+              production_report = [];
+              production_report.push({});
+            }
+            production_report = production_report[0];
+
+            if (ingredient in report) {
```

Comment:
This can be written like:
`production_report[ingredient] = (ingredient in report) ? report[ingredient] : 0;`

Totally being anal